src/App.tsx

```diff
@@ -1,34 +1,135 @@
+} from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
+import { debounce, filterTodos } from './helpers';
+import { Todo } from './types/Todo';
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos, getUser } from './api';
+import { Modal } from './components/Modal';
+import { User } from './types/User';
+import { LoadingError } from './components/LoadingError';
 
 export const App: React.FC = () => {
+  const filterByOptions = useMemo(() => [
+    {
+      title: 'All',
+      value: 'all',
+    },
+    {
+      title: 'Active',
+      value: 'active',
+    },
+    {
+      title: 'Completed',
+      value: 'completed',
+    },
+  ], []);
+
+  const [loading, setLoading] = useState(false);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todo, setTodo] = useState<Todo | null>(null);
+  const [user, setUser] = useState<User | null>(null);
+  const [selectedTodoId, setSelectedTodoId] = useState(0);
+  const [filterBy, setFilterBy] = useState(filterByOptions[0].value);
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [isOpenModal, setIsOpenModal] = useState(false);
+  const [hasLoadingError, setHasLoadingError] = useState(false);
+
+  const handleSelect = useCallback(
+    (value: string) => setFilterBy(value), [filterBy],
+  );
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, 1000), []);
+
+  const handleSearch = useCallback(
+    (value: string) => {
+      setQuery(value);
+      applyQuery(value);
+    }, [],
+  );
+
+  useEffect(() => {
+    getTodos()
+      .then(data => setTodos(data))
+      .catch(() => setHasLoadingError(true));
+  }, []);
+
+  const visibleTodos = useMemo(() => filterTodos(todos, filterBy, appliedQuery),
+    [todos, appliedQuery, query, filterBy]);
+
+  const closeModal = useCallback(() => {
+    setIsOpenModal(false);
+    setSelectedTodoId(0);
+    setUser(null);
+  }, [isOpenModal, user]);
+
+  const showTodoInfo = useCallback((currentTodo: Todo) => {
+    setLoading(true);
+    setIsOpenModal(true);
+    setTodo(currentTodo);
+    setSelectedTodoId(currentTodo.id);
+
+    getUser(currentTodo.userId)
+      .then(data => {
+        setLoading(false);
+        setUser(data);
+      })
+      .catch(() => setLoading(false));
+  }, [todo, user]);
+
   return (
     <>
       <div className="section">
         <div className="container">
           <div className="box">
             <h1 className="title">Todos:</h1>
 
-            <div className="block">
-              <TodoFilter />
-            </div>
+            {(!todos.length && !hasLoadingError) && <Loader />}
 
-            <div className="block">
-              <Loader />
-              <TodoList />
-            </div>
+            {todos.length > 0 && (
```

Comment:
```suggestion
            {todos.length && (
```

Reply:
<img width="861" alt="Screenshot 2023-03-21 at 19 12 28" src="https://user-images.githubusercontent.com/36309787/226688713-845e584a-59b0-49ee-80d0-4a71ceb29b93.png">

todos.length is displaying on the page

Reply:
!!todos.length you can try to use :)